modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ContravariantLaws.kt

```diff
@@ -2,32 +2,37 @@ package arrow.test.laws
 
 import arrow.Kind
 import arrow.core.compose
+import arrow.core.extensions.eq
 import arrow.test.generators.functionAToB
 import arrow.typeclasses.Contravariant
 import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
 import io.kotlintest.properties.Gen
 import io.kotlintest.properties.forAll
 
 object ContravariantLaws {
 
-    fun <F> laws(CF: Contravariant<F>, cf: (Int) -> Kind<F, Int>, EQ: Eq<Kind<F, Int>>): List<Law> =
-        InvariantLaws.laws(CF, cf, EQ) + listOf(
-            Law("Contravariant Laws: Contravariant Identity") { CF.identity(cf, EQ) },
-            Law("Contravariant Laws: Contravariant Composition") { CF.composition(cf, EQ) }
-        )
+  fun <F> laws(CF: Contravariant<F>, G: Gen<Kind<F, Int>>, EQK: EqK<F>): List<Law> {
```

Comment:
this would need to GenK's 
one for Kleisli and one for AndThen. i had a try without success on AndThen, maybe you could help here. There is already a genK for Kleisli in the test.

there are some further references from within the tests. i would then make them internal

Reply:
Yes, I forgot those. Let's leave it then